packages/kbn-reporting/public/share/share_context_menu/register_pdf_png_modal_reporting.tsx

```diff
@@ -0,0 +1,551 @@
+      const dimensions = { height, width };
+
+      const decoratedJobParams = apiClient.getDecoratedJobParams({
+        ...getJobParams(apiClient, jobProviderOptions, 'printablePdfV2')(),
+        layout: { id: 'preserve_layout', dimensions },
+        objectType,
+        title: sharingData.title,
+      });
+
+      return apiClient
+        .createReportingJob('printablePdfV2', decoratedJobParams)
+        .then(() => {
+          toastsReport.addSuccess({
+            title: intlReport?.formatMessage(
+              {
+                id: 'reporting.share.modalContent.successfullyQueuedReportNotificationTitle',
+                defaultMessage: 'Queued report for {objectType}',
+              },
+              { objectType }
+            ),
+            text: toMountPoint(
+              <FormattedMessage
+                id="reporting.share.modalContent.successfullyQueuedReportNotificationDescription"
+                defaultMessage="Track its progress in {path}."
+                values={{
+                  path: (
+                    <a href={apiClient.getManagementLink()}>
+                      <FormattedMessage
+                        id="reporting.share.publicNotifier.reportLink.reportingSectionUrlLinkLabel"
+                        defaultMessage="Stack Management &gt; Reporting"
+                      />
+                    </a>
+                  ),
+                }}
+              />,
+              { theme, i18n: i18nStart }
+            ),
+            'data-test-subj': 'queueReportSuccess',
+          });
+          if (onClose) {
+            onClose();
+          }
+        })
+        .catch((error: any) => {
+          toastsReport.addError(error, {
+            title: intlReport!.formatMessage({
+              id: 'reporting.share.modalContent.notification.reportingErrorTitle',
+              defaultMessage: 'Unable to create report',
+            }),
+            toastMessage: (
+              // eslint-disable-next-line react/no-danger
+              <span dangerouslySetInnerHTML={{ __html: error.body.message }} />
+            ) as unknown as string,
+          });
+        });
+    };
+
+    const generateReportPDFForPrinting = ({
+      intl: intlReport,
+      toasts: toastsReport,
+    }: ShareContext) => {
+      const el = document.querySelector('[data-shared-items-container]');
+      const { height, width } = el ? el.getBoundingClientRect() : { height: 768, width: 1024 };
+      const dimensions = { height, width };
+
+      const decoratedJobParams = apiClient.getDecoratedJobParams({
+        ...getJobParams(apiClient, jobProviderOptions, 'printablePdfV2')(),
+        layout: { id: 'print', dimensions },
+        objectType,
+        title: sharingData.title,
+      });
+      return apiClient
+        .createReportingJob('printablePdfV2', decoratedJobParams)
+        .then(() => {
+          toastsReport.addSuccess({
+            title: intlReport?.formatMessage(
```

Comment:
We can use the `intl` and `toasts` passed to `getShareMenuItems`
```suggestion
          toasts.addSuccess({
            title: intl.formatMessage(
```